src/glyph/pen.rs

```diff
@@ -0,0 +1,532 @@
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashSet;
+use std::hash::{Hash, Hasher};
+
+use crate::error::ErrorKind;
+use crate::glyph::{
+    Advance, AffineTransform, Anchor, Component, Contour, ContourPoint, GlifVersion, Glyph,
+    GlyphName, Guideline, Image, Outline, Plist, PointType,
+};
+use crate::shared_types::Identifier;
+
+#[derive(Debug)]
+pub struct Pen {
```

Comment:
I want docs! What is this, when should you use it, etc. Does this even need to be public?